src/utils/log.js

```diff
@@ -0,0 +1,10 @@
+function warn(content) {
+  if (process.env.NODE_ENV !== 'production') {
+    const message = (typeof content === 'function') ? content() : content;
+    if (message) {
+      console.warn(message); // eslint-disable-line no-console
+    }
+  }
+}
+
+export default { warn }
```

Comment:
NOTE: IMO this would be more readable as `export function warn( // ...`

instead of declaration and export (using default?) in separate places